static/js/util/api.js

```diff
@@ -33,6 +33,29 @@ export function csrfSafeMethod(method: string): boolean {
   return /^(GET|HEAD|OPTIONS|TRACE)$/.test(method);
 }
 
+export function fetchWithCSRF(path: string, init: Object = {}): Promise<*> {
```

Comment:
We should have `fetchJSONWithCSRF` call this code if that's feasible to reduce code duplication